src/DesktopTests/UIAutomation/CustomObjects/RegistrarTests.cs

```diff
@@ -1,45 +1,86 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Axe.Windows.Core.CustomObjects;
+using Axe.Windows.Core.CustomObjects.Converters;
+using Axe.Windows.Core.Enums;
 using Axe.Windows.Desktop.UIAutomation.CustomObjects;
 using Interop.UIAutomationCore;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
 using System;
 
 namespace Axe.Windows.DesktopTests.UIAutomation.CustomObjects
 {
     [TestClass]
     public class RegistrarTests
     {
-        // Note: We normally avoid Guid.NewGuid() to make tests more consistently reproducible. This class
-        // intentionally uses Guid.NewGuid to reduce the risk of the UIA state contamination between test runs.
-
-        [TestMethod]
-        [Timeout(1000)]
-        public void RegisterCustomProperty_ReturnsNonZeroValue()
+        [TestMethod, Timeout(1000)]
+        public void RegisterCustomProperty_CorrectDataFlow()
         {
-            using (Registrar r = new Registrar())
+            Guid testGuid = new Guid("312F7536-259A-47C7-B192-AA16352522C4");
+            string testName = "CommentReplyCount";
+            CustomProperty testConfig = new CustomProperty { Guid = testGuid, ProgrammaticName = testName, ConfigType = "int" };
```

Comment:
Some naming observations:
- `testConfig` is a Property, not a Config. Would `testProperty` be better?
- `testInfo` represents a value that we expect the production code to generate--maybe `expectedPropertyInfo` or similar?